src/citrine/exceptions.py

```diff
@@ -85,16 +85,16 @@ class WorkflowNotReadyException(RetryableException):
     pass
 
 
-class PollingTimeoutError(RetryableException):
+class PollingTimeoutError(NonRetryableException):
     """Polling for an asynchronous result has exceeded the timeout."""
 
     pass
 
 
-class JobFailureError(RetryableException):
+class JobFailureError(NonRetryableException):
     """The asynchronous job completed with the given failure message."""
 
-    def __init__(self, message: str, job_id: UUID, failure_reasons: List[str]):
+    def __init__(self, message: str, *, job_id: UUID, failure_reasons: List[str]):
```

Comment:
Change of suggestion following a conversation with @maxhutch 
```suggestion
    def __init__(self, *, message: str, job_id: UUID, failure_reasons: List[str]):
```